src/main/scala/se/nimsa/sbx/box/BoxDAO.scala

```diff
@@ -356,6 +356,13 @@ class BoxDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: ExecutionCont
     boxQuery.drop(startIndex).take(count).result
   }
 
+  def listPendingOutgoingTransactionsForBox(boxId: Long): Future[Seq[OutgoingTransaction]] = db.run {
+    outgoingTransactionQuery
+      .filter(_.boxId === boxId)
+      .filter(_.status === (WAITING: TransactionStatus))
+      .result
+  }
+
```

Comment:
This is used to detect new outgoing transactions that need to be sent. `BoxPushActor` calls this periodically and creates a push pipeline for each waiting transaction found.